decisions/2022-12-12_patient_and_program_transfer.md

```diff
@@ -0,0 +1,128 @@
+A patient may change the clinic where the program is treated.
+Thus, we need to be able to record the program store affiliation.
+Potentially a program can be treated at multiple stores.
+It's probably also useful to have a clinic transfer history.
+
+There are three types of transfers:
+
+- External in transfers: program transfer from an external clinic to mSupply
+- External out transfers: program is not manage by mSupply any more
+- Internal transfers, a program is transferred to a different store/clinic within mSupply
+
+**Patient visit**
+A patient might have a program encounter at another store.
+
+Patient Visit Permissions:
+
+- Required patient access during a visit should be minimal.
+- Practitioner needs access to base patient information
+- A visit should show up at the patient's clinic (and resets the "lost to follow up" status)
+
+## Options
+
+### Option 1
+
+**Patient editing:**
+
+- When changing a patient document the patient `name` is updated and pushed to central
+- For this to work we need to allow client to edit patients at stores that are not the patient's home store.
+  (Because there is no fixed store affiliation for programs)
+- When opening a patient document, merge back potential changes from the `name` table to the patient document.
+  This makes it possible to integrate edits of the patient's name table without help from the central server.
+
+**Patient lookup/visibility/sync:**
+
+- Mainly as done in current mSupply
+- Patient documents are synced through the `name_store_join`, i.e. a patient's documents are synced to stores associated with the patient. (in omSupply just trigger sync to fetch patient document data?)
+
+**Access permissions**
+As all mSupply permissions, user program permissions are defined for a specific store.
+This means a user can only access documents at a certain store when having the matching store permissions.
+
+**Patient transfers**
+
+To keep track of the current active store for a program a new `program_store_join` table is used.
+For example (details to be worked out):
+
+```typescript
+{
+  id: string,
+  patient_id: string,
+  program_id: string,
+  store_id: string,
+  // Primary store for this program (might not be needed?)
+  is_main_store: boolean,
+  // Row from where the patient has been transferred
+  transfer_in_id?: string,
+  // Details about external transfers
+  transfer_in_external?: string,
+  // transfer in time
+  transfer_in_datetime: Date,
+  // details columns about where the patient has been transferred
+  transfer_out_id?: string,
+  // Details about external transfers
+  transfer_out_external?: string,
+  // If set the row is "inactive" because the patients has been transferred out
+  transfer_out_datetime?: Date,
+}
+```
+
+This table can also be used to derive a transfer history.
+Furthermore, this table can also be used to decide if further edits are allowed, e.g. if the patient has been transferred out and thus a program is "inactive" at a certain store.
```

Comment:
Is this still true ? I thought we wouldn't restrict editing of patient to home store only ?

Also the transfer table (program_store_join) can potentially just have `id`, `patient_id`, `program_id`, `datetime`, 'name_id` and `external_name`. `name_id` optional (only if facility exists in mSupply) can be either name for a store or store less name, and `external_name` if facility does not exist in mSupply.

I think above is enough to capture history, and if it's considered patient data, can sync to name_store_join stores

Reply:
thanks, I updated the document:
- `program_store_join` -> `patient_transfer`
- only have external_out_name, external_in_name if transferred out or in
- Removed the comment. Was more about to be able to show that the patient is not active at the current store anymore.

Reply:
> Removed the comment. Was more about to be able to show that the patient is not active at the current store anymore.

Oki, I think potentially having a boolean to identify `current` record in that table might help joining `patient_transfer` table to say name table